sidekick_core/lib/sidekick_core.dart

```diff
@@ -68,8 +73,8 @@ SidekickCommandRunner initializeSidekick({
     repository: repo,
     mainProject: mainProject,
     workingDirectory: Directory.current,
-    flutterSdk: flutterSdkPath == null ? null : Directory(flutterSdkPath),
-    dartSdk: dartSdkPath == null ? null : Directory(dartSdkPath),
+    flutterSdk: _resolveSdkPath(flutterSdkPath),
+    dartSdk: _resolveSdkPath(dartSdkPath),
```

Comment:
You could use the same method as for the `mainProject`:

```dart
  mainProject =
        DartPackage.fromDirectory(repo.root.directory(mainProjectPath));
```

```dart
flutterSdk = Directory(repo.root.directory(flutterSdkPath));
```

Plus, add the `isAbsolute` check